### PR TITLE
fix: fix sentry traces

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **fix**

This PR resolves [MOPLAT-119]

### Description

We had a couple issues, the main issue is our release name passed with errors is not the same as what we set when uploading source maps due to an issue with SDK.  The other one is because of how our js is bundled as part of emission pod on iOS side if we don't do things in the correct order (updating the sentry release name in json, installing pods, building iOS app) we get an old value for sentry release name in the built app, which breaks source map matching and stack traces. I tried to get this working for quite a while which is why so many commits before eventually giving up and just deciding it is better to have the release name in 2 places (app and fastfile) then have this confusing order requirement that can easily break again.

- Removes sentryReleaseName and update logic from json and instead just uses same format in app and fastfile
- Passes dist value to sentry init to workaround issue with sentry not respecting release names: https://github.com/getsentry/sentry-react-native/issues/805
- Upgrades sentry to latest version
- Removes patch that is now unnecessary with update
- Fixes the sentry init check, this was reversed but was working because feature flag value was defaulting to false

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Fix sentry stack traces in errors - brian

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[MOPLAT-119]: https://artsyproduct.atlassian.net/browse/MOPLAT-119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 64d8a7c92cbfbae3067b5d65ca64f15c978ffd39
</p>

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Fix sentry stack traces in errors - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3a4b672f26388eaa7626670e62ea710f380d5397
</p>

File: src/lib/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/Components/ArtistAutosuggestResults.tsx [hunk 1]
```diff
@@ -70,7 +72,9 @@ const ArtistAutosuggestResultsFlatList: React.FC<{
   results.current = latestResults || results.current
 
   const nodes: ArtistAutosuggestResult[] = useMemo(
-    () => results.current?.results?.edges?.map((e, i) => ({ ...e?.node!, key: e?.node?.href! + i })) ?? [],
+    () =>
+      results.current?.results?.edges?.map((e, i) => ({ ...e?.node!, key: e?.node?.href! + i })) ??
+      [],
```
Comment (olerichter00) [inline src/lib/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/Components/ArtistAutosuggestResults.tsx:77]: Maybe we can use `extractNodes` here to simplify 🤔

File: src/lib/ErrorReporting.ts [hunk 2]
```diff
@@ -1,17 +1,28 @@
 import * as Sentry from "@sentry/react-native"
 import { useEffect } from "react"
+import { Platform } from "react-native"
 import Config from "react-native-config"
-import { sentryReleaseName } from "../../app.json"
+import { getBuildNumber, getVersion } from "react-native-device-info"
 import { GlobalStore, useFeatureFlag } from "./store/GlobalStore"
 
+// important! this much match the release version specified
```
Comment (pvinis) [inline src/lib/ErrorReporting.ts:8]:
```suggestion
// important! this must match the release version specified
```

File: HACKS.md [hunk 3]
```diff
@@ -260,16 +260,6 @@ Once storybook is upgraded to a version that does not use the removed `Cancellab
 
 We get an error like here, and that is the solution. https://github.com/DefinitelyTyped/DefinitelyTyped/issues/47166#issuecomment-685738545
 
-## typings @sentry/react-native
```
Comment (pvinis) [inline HACKS.md:263]: WOOP! for every hack removed, a beautiful butterfly is born.

**Review by olerichter00 — APPROVED** [review]
Looks good to me 👍

**Review by pvinis — APPROVED** [review]
very nice! i dont have the time to test this today, but i think it would be way easier to test this on beta anyway. is that right brian?

File: src/lib/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/Components/ArtistAutosuggestResults.tsx (hunk 1 above)
Comment (pvinis) [reply src/lib/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/Components/ArtistAutosuggestResults.tsx:77]: while that is true, i think this change is just prettier running, not brian changing things. maybe we can use `extractNodes` on a separate pr?

Comment (brainbicycle) [reply src/lib/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/Components/ArtistAutosuggestResults.tsx:77]: yeah these are prettier changes, I am not sure if my setup is different or something but I didn't intentionally change these, I can remove

**Comment by brainbicycle** [comment]
> very nice! i dont have the time to test this today, but i think it would be way easier to test this on beta anyway. is that right brian?

yeah I have been testing in betas since it relies on the sourcemap upload and store builds that seems like easiest way to me

File: src/lib/ErrorReporting.ts [hunk 4]
```diff
@@ -1,17 +1,28 @@
...
+  return prefix + "-" + version + "-" + buildNumber
+}
+
 export const setupSentry = (
   props: Partial<Sentry.ReactNativeOptions> = {},
-  captureExceptions = !__DEV__
+  captureExceptions: boolean
 ) => {
-  if (!captureExceptions && Config.SENTRY_DSN) {
+  if (captureExceptions && Config.SENTRY_DSN) {
```
Comment (MounirDhahri) [inline src/lib/ErrorReporting.ts:21]: thanks for the fix

Comment (MounirDhahri) [reply src/lib/ErrorReporting.ts:21]: no idea how this working

**Comment by brainbicycle** [comment]
> LGTM 🎉 Thanks Brian
> On a side note, I really hate the 120 to 100 change. Half of the changes in PRs the previous days are prettier changes 😄 and the git blame is no longer reliable

I wonder if we should just bite the bullet and do a large PR with all the changes, then at least the blame is limited to one commit.

**Comment by MounirDhahri** [comment]
Good idea, I will do it

**Pull request merged**